worlds/banjo_tooie/__init__.py

```diff
@@ -0,0 +1,1094 @@
+            (itemName.ENEST, self.options.egg_nests_weight * (2 if self.options.nestsanity else 1)),
+            (itemName.FNEST, self.options.feather_nests_weight * (2 if self.options.nestsanity else 1)),
+            # Intentionally leaving NONE as last;
+            # because self.random.choices might always choose last if all weights are 0
+            (itemName.NONE, self.options.big_o_pants_weight)
+        ]
+
+        if self.traps_in_pool < self.options.max_traps:
+            weights = trap_weights + filler_weights
+        else:
+            weights = filler_weights
+
+        names, actual_weights = zip(*weights)
+
+        if sum(actual_weights) == 0:
+            actual_weights = (*actual_weights[:-1], 1)
+
+        return self.random.choices(names, actual_weights, k=1)[0]
+
+    def banjo_pre_fills(self, itemNameOrGroup: str, group: str | None, useGroup: bool) -> None:
+        if useGroup:
+            for group_name, item_info in self.item_name_groups.items():
+                if group_name == itemNameOrGroup:
+                    for name in item_info:
+                        item = self.create_item(name)
+                        banjoItem = all_item_table.get(name)
+                        # self.multiworld.get_location(banjoItem.defualt_location, self.player).place_locked_item(item)
+                        location = self.get_location(banjoItem.default_location)
+                        location.place_locked_item(item)
+        else:
+            for name, id in self.location_name_to_id.items():
+                item = self.create_item(itemNameOrGroup)
+                if self.location_name_to_group[name] == group:
+                    # self.multiworld.get_location(name, self.player).place_locked_item(item)
+                    location = self.get_location(name)
+                    location.place_locked_item(item)
+
+    @classmethod
+    def stage_write_spoiler(cls, world, spoiler_handle):
+        entrance_hags = {
+            regionName.MT: regionName.MTE,
+            regionName.GM: regionName.GGME,
+            regionName.WW: regionName.WWE,
+            regionName.JR: regionName.JRLE,
+            regionName.TL: regionName.TDLE,
+            regionName.GIO: regionName.GIE,
+            regionName.HP: regionName.HFPE,
+            regionName.CC: regionName.CCLE,
+            regionName.CK: regionName.CKE,
+            regionName.MTBOSS: regionName.MTTT,
+            regionName.GMBOSS: regionName.CHUFFY,
+            regionName.WWBOSS: regionName.WW,
+            regionName.JRBOSS: regionName.JRLC,
+            regionName.TLBOSS: regionName.TLTOP,
+            regionName.GIBOSS: regionName.GI1,
+            regionName.HPFBOSS: regionName.HP,
+            regionName.HPIBOSS: regionName.HP,
+            regionName.CCBOSS: regionName.CC,
+        }
+        bt_players = world.get_game_players(cls.game)
+        spoiler_handle.write('\n\nBanjo-Tooie ({})'.format(BanjoTooieWorld.version))
```

Comment:
```suggestion
        spoiler_handle.write(f"\n\nBanjo-Tooie (V{cls.world_version.as_simple_string()})")
```
Why not use an f-string here? Also here's an example of how you can use `world_version`.